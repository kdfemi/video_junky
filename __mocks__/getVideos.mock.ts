export const videoMock =  {
    "kind":"youtube#searchResult",
    "etag":"2XRyyyjfe4AsHOsGqmKcuAMyUCM",
    "id":{
       "kind":"youtube#video",
       "videoId":"SO1th-Oh39I"
    },
    "snippet":{
       "publishedAt":"2023-10-08T18:00:15Z",
       "channelId":"UCjIuqG1hCxUdLLxOD_uR0vQ",
       "title":"Joh John Florence And Friends At Rocky Point (4K Raw)",
       "description":"Filmed on October 6, 2023 at Rocky Point on the North Shore of Oahu in Hawai'i. Watch as John John Florence goes for a free ...",
       "thumbnails":{
          "default":{
             "url":"https://i.ytimg.com/vi/SO1th-Oh39I/default.jpg",
             "width":120,
             "height":90
          },
          "medium":{
             "url":"https://i.ytimg.com/vi/SO1th-Oh39I/mqdefault.jpg",
             "width":320,
             "height":180
          },
          "high":{
             "url":"https://i.ytimg.com/vi/SO1th-Oh39I/hqdefault.jpg",
             "width":480,
             "height":360
          }
       },
       "channelTitle":"Surfers of Hawaii",
       "liveBroadcastContent":"none",
       "publishTime":"2023-10-08T18:00:15Z"
    }
} as const;

export const getVideosResponseMock = {
    totalSize: 50,
    page: 1,
    size: 10,
    pageCount: 100,
    pages: 5,
    results: [videoMock]
}

export const getVideosResponseEmptyMock = {
    totalSize: 0,
    page: 1,
    size: 10,
    pageCount: 0,
    pages: 0,
    results: []
}