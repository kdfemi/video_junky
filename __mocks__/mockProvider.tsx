/// Mock Provider is a wrapper around @testing-library/react render method
/// This provides us away to mock Provider used app wide

import React, { PropsWithChildren } from 'react';
import { RenderOptions, render as rtlRender } from '@testing-library/react';

type CustomRenderOptions =  RenderOptions;

function render(
    ui: React.ReactElement,
    {
      ...renderOptions
    } = {} as CustomRenderOptions,
  ) {
    function Wrapper({ children }: PropsWithChildren) {
      return <>{children}</>;
    }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}


export * from '@testing-library/react';

export { render };