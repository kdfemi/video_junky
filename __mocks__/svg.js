/// Fallback support for SVG during testing

// eslint-disable-next-line import/no-anonymous-default-export
export default 'svgr-url';
export const ReactComponent = 'div'

