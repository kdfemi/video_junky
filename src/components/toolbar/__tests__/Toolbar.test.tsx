import '@testing-library/jest-dom';
import Toolbar from ".."
import { render, screen } from "../../../../__mocks__/mockProvider"

describe('Toolbar', () => {
    it('Should be in document', () => {
        render(<Toolbar />)
        const toolbar = screen.getByTestId('toolbar');
        expect(toolbar).toBeInTheDocument()
    })
    it('Should match snap shot', () => {
        render(<Toolbar />)
        const toolbar = screen.getByTestId('toolbar');
        expect(toolbar).toMatchSnapshot()
    })
})