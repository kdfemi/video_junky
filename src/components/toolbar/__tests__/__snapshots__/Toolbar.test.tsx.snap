// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar Should match snap shot 1`] = `
<header
  class="flex items-center pt-8 px-6 gap-x-6 justify-between "
  data-testid="toolbar"
>
  <div
    class="text-2xl bg-logo-gradient block text-transparent bg-clip-text font-roboto font-bold"
  >
    JUnkie
  </div>
  <nav
    class="flex items-center nav"
  >
    <ul
      class="flex items-center justify-center gap-x-6 text-base max-md:hidden font-roboto"
    >
      <a
        class="nav__item"
        href="#"
      >
         
        Videos
      </a>
      <a
        class="nav__item"
        href="#"
      >
         
        Discover
      </a>
      <a
        class="nav__item"
        href="#"
      >
         
        Trending
      </a>
    </ul>
  </nav>
  <div>
    <div
      class="w-5 h-5 rounded-sm bg-gradient-conic to-junky-green from-junky-yellow"
    />
  </div>
</header>
`;
